src/string.rs

```diff
@@ -1 +1,289 @@
+fn sa_naive<T: Ord>(s: &[T]) -> Vec<usize> {
+    let n = s.len();
+    let mut sa: Vec<usize> = (0..n).collect();
+    sa.sort_by(|&(mut l), &(mut r)| {
+        if l == r {
+            return std::cmp::Ordering::Equal;
+        }
+        while l < n && r < n {
```

Comment:
This supposed to be faster than `s[l..].cmp(&s[r..])`, I guess?

Reply:
Probably in the original ACL. Not sure if it is still faster than `s[l..].cmp(&s[r..])` in Rust.
Trying some benchmarks...

Reply:
https://play.rust-lang.org/?version=stable&mode=release&edition=2018&gist=221f1644cdef92f2f05ef28f7ba2f85d
In this particular example manually inlining slice comparisons is slightly faster on AtCoder's Custom Test:
```
method: naive_0 15000 times => 1.718785665s (hash = 6269349985963744704)
method: naive_1 15000 times => 2.029751916s (hash = 6269349985963744704)
method: naive_0 15000 times => 1.709326945s (hash = 6269349985963744704)
method: naive_1 15000 times => 2.01106749s (hash = 6269349985963744704)
```

Reply:
Oh great! I wasn't expect this much difference!